src/app/group/group-add-edit/group-add-edit.component.ts

```diff
@@ -11,68 +12,180 @@ import { filter } from 'rxjs';
 export class GroupAddEditComponent implements OnInit {
 
   addGroupForm !: FormGroup;
-  groupTypeArray: string[] = ['Trip', 'Couple', 'Other'];
+  addMembersForm !: FormGroup;
+
   isGroupAddActive: Boolean = false;
-  getActivatedRouteParam : String = '';
+  getActivatedRouteParam: String = '';
+  isGroupCreated: boolean = false;
+
+  GroupResult !: GroupResult;
+  userProfile!: UserProfile;
 
   // Constructor
-  constructor( private router: Router, private activatedRoute: ActivatedRoute){};
+  constructor(
+    private router: Router,
+    private activatedRoute: ActivatedRoute,
+    private groupsService: GroupsService
+  ) { }
 
   // ngOnInit method
   ngOnInit(): void {
+    /** Form group for add group page */
     this.addGroupForm = new FormGroup({
       groupName: new FormControl('', [Validators.required]),
-      shareLink: new FormControl(''),
-      groupType: new FormControl('Home'),
+      about: new FormControl(''),
       groupMember: new FormArray([
-        this.memberNameAndEmailForm()
+        new FormGroup({
+          userId: new FormControl('3a0ba79a-0f40-0dd1-4e91-9ed501777180')
+        })
       ]),
     })
 
-    // get activatedRoute parameter using observable
-    this.activatedRoute.params.subscribe((param) =>{
+    /** form group add members to group */
+    this.addMembersForm = new FormGroup({
+      groupMembers: new FormArray([
+        this.addMembersToGroupForm()
+      ]),
+    })
+
+    /** get activatedRoute parameter using observable */
+    this.activatedRoute.params.subscribe((param) => {
       this.getActivatedRouteParam = param['routerParam'];
-      console.log(this.getActivatedRouteParam)
-      if(this.getActivatedRouteParam === undefined ){
+
+      if (this.getActivatedRouteParam === undefined) {
         this.isGroupAddActive = true;
       }
-      else{
+      else {
         this.isGroupAddActive = false;
       }
     })
-  }
 
-  // Add member name and member email formgroup in add group form
-  memberNameAndEmailForm(): FormGroup{
+    if (this.getActivatedRouteParam != undefined) {
+      this.addGroupForm.patchValue({
+        groupName: 'trip',
+        about: 'this is group for testing',
+        groupMember: [{
+          userId: 'admin'
+        }]
+      })
+    }
+
+
+    /** On click of group edit button it will set current details of group to textbox */
+    if (this.getActivatedRouteParam != undefined) {
+      const data = this.groupsService.groupList[Number(this.getActivatedRouteParam)];
+      this.addGroupForm.patchValue({
+        groupName: data.name,
+        about: data.about,
+        groupMember: [{
+          userId: data.createdBy
+        }]
```

Comment:
Semi colon

Reply:
It showing error when i try to add semi colon here.

Reply:
you might add ing it on wrong place